### PR TITLE
fix: parameter linking and deterministic par names

### LINKED ISSUE #8: Non deterministic generated fit parameter names from the expert system
Fit parameter names generated by the expertsystem are not deterministic.

**To Reproduce**
Just run the quickstart jupyter notebook a couple of times. Eventually it fails, because the fit parameters may have changed.

**Expected behavior**
Maybe there can be a more deterministic behavior implemented.

### LINKED ISSUE #9: Expert system cannot link amplitude coefficients to full extend
The full helicity amplitude generated by the expert system is currently only able to link coefficients of sequential amplitudes with other ones based on the parity conservation. 

This is not a general implementation and has to be improved. It seems more flexible to perform coefficient matching on a partial amplitude level and combine them later on into a single coefficient for a sequential amplitude, if possible. It is not possible to combine all coefficients of a sequential amplitude, if all coefficients are linked to coefficients of other sequential amplitudes.

When redesigning this, it should also be easier to automatically set appropriate coefficients to fixed, to have a well formed fitting problem (otherwise no convergence).

### PR DESCRIPTION
**Pull request by spflueger**
This smaller PR fixes several issues:

* Closes #8 
* Closes #9 
* Fixes the `_yaml_adapter` to generate the correct recoil system properties.

### CODE REVIEW

**Comment by spflueger** [comment]
I could not verify that the `Prefactors` are inserted correctly into the recipe files so far. I think none of the tests we have actually use prefactors... so I will implement a small test for that tomorrow. Nevertheless this can be reviewed already.

**Comment by redeboer** [comment]
I think at this stage we may want to add a JSON schema for the output YAML schema. I think I have that somewhere in a branch (it's quite a complicated one because of the recursive intensity section), so I'll also upload that tomorrow (after checking if it works correctly).

**Comment by redeboer** [comment]
> I think at this stage we may want to add a JSON schema for the output YAML recipe file.

Okay JSON schema for YAML recipe output has been added. However, I still need to implement a `jsonschema.validate` statement, but for that I would need to do something like [this](https://stackoverflow.com/a/53968771), because we import external JSON schema files (single source of truth).

**Comment by spflueger** [comment]
> > I think at this stage we may want to add a JSON schema for the output YAML recipe file.
> 
> Okay JSON schema for YAML recipe output has been added. However, I still need to implement a `jsonschema.validate` statement, but for that I would need to do something like [this](https://stackoverflow.com/a/53968771), because we import external JSON schema files (single source of truth).

I fixed this with 84d48698623d36f5c399e37c7a8ef987c2897ab3

**Comment by spflueger** [comment]
There were still some problems with the parameter name generation and correct insertion of the `PreFactor` in the output yaml. These have been fixed now! I still want to add some tests regarding the prefactor, then this PR is finished.

Btw, I don't get why the json files are reformatted by prettier in my env. Were they not previously formatted? (see 84d48698623d36f5c399e37c7a8ef987c2897ab3)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -39,6 +39,7 @@ repos:
     rev: 2.0.5
     hooks:
       - id: prettier
+        language_version: 12.18.2 # prettier does not specify node correcly
```
[the remark this answers is not in the document]
Comment (redeboer) [reply .pre-commit-config.yaml:42]: I tried out some of the [Prettier config options](https://prettier.io/docs/en/configuration.html) but they don't help either.

File: expertsystem/schemas/yaml/amplitude-model.json [hunk 2]
```diff
@@ -0,0 +1,542 @@
+{
+  "$id": "amplitude-model.json",
```
[the remark this answers is not in the document]
Comment (spflueger) [reply expertsystem/schemas/yaml/amplitude-model.json:2]: Hmm I'm not sure if this is the correct way. I saw he is really querying this URL for the lookup. Therefore if a file is not yet commited to the repo that url does not exist -> problem

File: .pre-commit-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (spflueger) [reply .pre-commit-config.yaml:42]: I got some error message that my node version should be at least 11 something or so. My local node version is actually higher than that. Pre-commit installs some own environments for all of the various hooks, and this nodejs env for prettier (and some other hook of ours as well) was just really old (node 8). For some reason the prettier pre-commit config does not specify it needs node > 11. Hence I overwrote the config here

File: expertsystem/schemas/yaml/amplitude-model.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (redeboer) [reply expertsystem/schemas/yaml/amplitude-model.json:2]: Yeah and besides, if you make changes, those changes won't appear in the master

[the remark this answers is not in the document]
Comment (spflueger) [reply expertsystem/schemas/yaml/amplitude-model.json:2]: True, so i dont quite understand why you would put an URL there unless its something stable you are referring to

File: .pre-commit-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (redeboer) [reply .pre-commit-config.yaml:42]: We still have the problem that we want Prettier formatting to be done through VSCode as well, or at least, we recommend Prettier as an extension. But the installation that the extension uses is not the same as the one used by pre-commit. Also, the VSCode Prettier extension and the pre-commit Prettier hook only share settings through the `.prettierrc` file. So the `language_version` option in this `.pre-commit-config.yaml` file won't be seen by the VSCode extension.

[the remark this answers is not in the document]
Comment (spflueger) [reply .pre-commit-config.yaml:42]: agreed

**Pull request merged**